### PR TITLE
Elasticsearch 1.4.2

### PR DESCRIPTION
**Pull request by jdeniau**
I tried to update to 1.4.2 (based on your 1.4.0 branch).

I now have 8 failing tests:
7 with this message : 
```
Failed asserting that exception of type "Elastica\Exception\ResponseException" matches expected exception "\Elastica\Exception\RuntimeException". Message was: "ClusterBlockException[blocked by: [FORBIDDEN/5/index read-only (api)];]" 
```

The last one is that ```Thread.sleep(100); ``` is [not authorized anymore now by ES](https://github.com/elasticsearch/elasticsearch/blob/master/src/test/java/org/elasticsearch/script/GroovySandboxScriptTests.java#L82) 

I do not know how to solve those problems :/

### CODE REVIEW

**Comment by ruflin** [comment]
@jdeniau Did you have any luck yet in fixing the issues?

**Comment by jdeniau** [comment]
No sorry I did not look at it this week.

**Comment by jdeniau** [comment]
I worked on it today. 
The main problem is that the "read_only" instruction seem to block all queries, (see https://github.com/elasticsearch/elasticsearch/pull/9203 ).

I am down to ~15 errors for now, I will commit changes soon.

**Comment by ruflin** [comment]
@jdeniau That looks really promising. Only 1 error left. Make sure to merge in master as there are also quite a few changes.

I also started to work on it but you are already much farther: https://github.com/ruflin/Elastica/tree/elasticsearch-1.4.0

File: lib/Elastica/Index/Settings.php [hunk 1]
```diff
@@ -118,7 +120,18 @@ public function setNumberOfReplicas($replicas)
...
+
+    /**
+     * getReadOnly
+     *
+     * @access public
+     * @return bool
+     */
+    public function getReadOnly()
+    {
+        return $this->get('blocks.write') === 'true'; // ES returns a string for this setting
```
Comment (ruflin) [inline lib/Elastica/Index/Settings.php]: This variable sounds so counterintuitive. But it is 100% correct. I also checked the ES docs.

**Comment by jdeniau** [comment]
The failing test is really weird.

Actually, it goes well on my local instance of ES.
Moreover the code on master is :
```php
// the document with the random score should have a score > 1, means it is the first result       
$result0 = $results[1]->getData();
```
But the first document is ```$results[0]```, not ```$results[1]``` (as said in the variable ```$result0``` too)

File: lib/Elastica/Index/Settings.php [hunk 2]
```diff
@@ -68,31 +68,33 @@ public function get($setting = '')
         $requestData = $this->request()->getData();
         $data = reset($requestData);
 
+        if (empty($data['settings']) || empty($data['settings']['index'])) {
+            return []; // index not found
```
Comment (ruflin) [inline lib/Elastica/Index/Settings.php]: Should we actually throw a not found exception here?

**Comment by jdeniau** [comment]
Sorry to push a lot but I have some differences between travis and my localhost instance :(

I took the responsability of removing the 5.3 support. I only did that to allow array short syntax, so it is easily revertable, but as the 5.3 version is not officially supported by the PHP developers, I think it is a good think to remove it.

But it's up to you, if you want to revert, I'll do it.

**Comment by ruflin** [comment]
@jdeniau No problem. The main priority is getting 1.4.2 running. To make it then compatible again with 5.3 if necessary is an other thing. Lets not focus on this right now.

**Comment by ruflin** [comment]
@jdeniau All green. Let me know when I can merge.

**Comment by jdeniau** [comment]
OK for me when the next build is OK.

**Comment by jdeniau** [comment]
It's OK for me ! :)

File: .travis.yml [hunk 3]
```diff
@@ -1,16 +1,19 @@
...
   - 5.4
   - 5.5
   - 5.6
 
-matrix:
-  allow_failures:
-    - php: 5.6
-
 env:
+  global:
```
Comment (im-denisenko) [inline .travis.yml]: This block(9-16) was deleted by me in #742.
Packages versions should be specified in ansible/es-playbook.yml now.
Actually, travis built your pull request with ES 1.3.4, as i can see from logs :)

**Comment by ruflin** [comment]
@jdeniau Can you please have a look at this?

File: .travis.yml (hunk 3 above)
Comment (ruflin) [inline .travis.yml]: Thanks for checking ...

**Comment by jdeniau** [comment]
Finally :sunglasses:
I am not proud of this commit https://github.com/jdeniau/Elastica/commit/07b56f63c5e2dd571b54e9162d00bdf53dc62ddc but it is the same kind of problem we have with the `readonly`.

Once merged, you can merge https://github.com/ruflin/Elastica/pull/735 (I updated it)

File: composer.json [hunk 4]
```diff
@@ -12,10 +12,10 @@
         }
     ],
     "require": {
-        "php": ">=5.3.3"
+        "php": ">=5.3.3",
```
Comment (im-denisenko) [inline composer.json]: With arrays new syntax, ">=5.4.0"

Comment (jdeniau) [inline composer.json]: I tried that but got a weird bug with composer in the tests: 
https://travis-ci.org/ruflin/Elastica/builds/47757241

File: composer.json [hunk 5]
```diff
@@ -12,10 +12,10 @@
...
+        "psr/log": "~1.0"
```
Comment (ruflin) [inline composer.json]: Why did you move this into required?

**Comment by ruflin** [comment]
@jdeniau I merged it in :-) That was a big pice of work. Thanks a lot.

Will you keep an eye on https://github.com/elasticsearch/elasticsearch/pull/9203 ? I assume as soon as it is merged in and release, it will break again the implementation.

**Comment by ruflin** [comment]
@jdeniau BTW: I think we forgot to update the README: https://github.com/ruflin/Elastica/blob/master/README.markdown

File: composer.json (hunk 4 above)
Comment (im-denisenko) [inline composer.json]: I've found a problem.
1. ubuntu 12.04 default php version is 5.3
2. ansible uses ```sudo``` internally for all steps
3. when you use [phpenv](https://github.com/CHH/phpenv) as travis does, ```php``` and ```sudo php``` is different versions of php
4. ```sudo composer``` can't satisfy ">=5.4" requirement, becase uses system php, which 5.3 
5. ansible wipe out all stderr from composer so it fails silently =/

Moreover, vagrant box is broken again, as there only one php version - 5.3

I'll fix this issue.

**Comment by jdeniau** [comment]
I created another PR https://github.com/ruflin/Elastica/pull/756 for this.

Thanks for the merge !

For the metadata bug on ES itself, it will be part of a new ElasticSearch version. I'll watch about it.

File: composer.json (hunk 4 above)
Comment (ruflin) [inline composer.json]: Thx. Looking forward to it.
